src/sqlite/query.rs

```diff
@@ -0,0 +1,122 @@
+use lunatic_sqlite_api::guest_api::sqlite_guest_bindings as bindings;
+use lunatic_sqlite_api::wire_format::{BindKey, BindList, BindPair, SqliteRow};
+
+use super::client::SqliteClient;
+use super::error::{SqliteCode, SqliteError, SqliteErrorExt};
+use super::value::Value;
+use crate::call_host_alloc;
+
+/// Trait for querying data and executing queries.
+pub trait Query {
+    /// Executes a query with no bindings.
+    fn query(&self, query: &str) -> QueryResult;
+    /// Prepares a query with bindings.
+    fn prepare_query(&self, query: &str) -> Statement;
+    /// Executes a query, ignoring any results.
+    fn execute(&self, query: &str) -> Result<(), SqliteError>;
+}
+
+impl Query for SqliteClient {
+    fn query(&self, query: &str) -> QueryResult {
+        self.prepare_query(query).execute()
+    }
+
+    fn prepare_query(&self, query: &str) -> Statement {
+        let id = unsafe { bindings::query_prepare(self.id(), query.as_ptr(), query.len() as u32) };
+        Statement {
+            id,
+            bindings: BindList(vec![]),
+        }
+    }
+
+    fn execute(&self, query: &str) -> Result<(), SqliteError> {
```

Comment:
The formatting is a bit inconsistent here I think

Reply:
We should move the clippy and formatting checks before the main test, like in the other repo.

Reply:
I don't think I got any lint/formatting issues here. What should be changed here exactly?

Reply:
I just meant that in the CI job of the VM we moved the lint/formatting checks before the actual tests. It can take a long time for the tests to compile, but formatting checks are much faster. The CI stops on the first job failed, so it can save you a few minutes of running if you have a formatting issue.

Reply:
Sorry I understood that part, I more so was asking about @SquattingSocrates' comment about formatting might not be consistent.